man/tallow.patterns.5.md

```diff
@@ -0,0 +1,91 @@
+## tallow.patterns
+
+Tallow pattern matching configuration files.
+
+
+## SYNOPSIS
+
+tallow(1) configures patterns to match journal entries and banning thresholds
+from JSON files.
+
+`/etc/tallow/*.json`
+`/usr/share/tallow/*.json`
+
+
+## DESCRIPTION
+
+tallow(1) configures patterns to match journal entries and banning thresholds
+from JSON files. Many JSON files can exist for logical grouping. The tallow(1)
+daemon will read all JSON files from configuration directories at startup.
+
+tallow(1) will operate with default pattern definitions
+in`/usr/share/tallow/*.json`. Users can add more patterns with their own JSON
+files under `/etc/tallow`. The default JSON files can be overridden by creating
+the same file under `/etc/tallow`.
+
+
+## FILE FORMAT 
+
+Pattern configuration files use the JavaScript Object Notation (JSON) format.
+
+The JSON needs to be two levels deep and all properties required. The root
+object is an array containing objects with a `filter` key and an `items` key. 
+
+* `filter` is a string that defines a field for filtering the journal file.
+  This helps make sure patterns are only matched to a subset of journal
+  entries. See systemd.journal-fields(7) for valid journal fields.
+
+* `items` is an array of objects that contains three elements: `ban`, `score`,
+  and `pattern`.
+
+   * `ban` is an integer that defines the number of seconds the originating IP
+     address should be banned for. The ban is applied immediately after a
+     `pattern` matches a journal entry.
+
+   * `score` is a double that defines a value to add to the accumulated "score"
+     of an originating IP address each time a journal entry matches
+     the `pattern`. If the combined score is > 1.0, tallow bans the originating
+     IP for the default time of 5 minutes. 
+
+   * `pattern` is a string that defines a Perl Compatible Regular Expressions
+     (PCRE) to match against the filtered journal entries. See
+     systemd.journal-fields(7) for valid journal fields.
+
+
+
+## EXAMPLES
+
+1. See the `/usr/share/tallow/sshd.json` file containing pattern matching
+   configurations for the `sshd` service.
+
+2. The JSON below defines patterns for blocking connections based on logs from
```

Comment:
@ahkok do you have a better out-of-tree example we can include?

Reply:
Just include one of the `sshd.json` ones ? I think those are the most well known and most easily understood values.

e.g. this one is great:
```
      {
        "ban": 10,
        "score": 0.2,
        "pattern": "MESSAGE=Invalid user .* from ([0-9a-z:.]+) port \\d+"
      },
```

Reply:
How about this one:

```
[
  {
    "filter": "SYSLOG_IDENTIFIER=nginx-mainline",
    "items": [
      {
        "ban": 10,
        "score": 0.2,
	"pattern": ".Primary script unknown. while reading response header from upstream, client: ([0-9a-z:.]+),"
      }
    ]
  }
]
```